### PR TITLE
12: dmsapi re and crs re

### PR DESCRIPTION
**Pull request by vifrrg**
- PEP8 formatting (partially)
- fixed url joining in DMS-related classes
- fixet `get_dms_log_url` bug (`1` component missing)

### CODE REVIEW

File: oc_cdtapi/ForemanAPI.py [hunk 1]
```diff
@@ -115,103 +105,101 @@ def get_owner_v1 (self, user_login):
...
-        logging.debug ('Passing to get_usergroup_id_v1')
-        return self.get_usergroup_id_v1 (group_name)
-
+        logging.debug('Reached get_usergroup_id_v2')
+        logging.debug('Passing to get_usergroup_id_v1')
+        return self.get_usergroup_id_v1(group_name)
 
     def _set_expiration(self):
         """
         A private method which sets the default expiration date (1 year from the current date)
```
Comment (ivansa-ru) [inline oc_cdtapi/ForemanAPI.py:159]: seems it's actually three months

File: oc_cdtapi/ForemanAPI.py [hunk 2]
```diff
@@ -301,861 +288,823 @@ def create_host_v2 (self, task, custom_json):
...
         :return: int
         """
-        logging.debug ('Reached get_hostgroup_id_v1')
-        hostgroups = self.get("hostgroups", headers = self.headers).json()["results"]
-        for hostgroup in hostgroups:
-            if hostgroup["name"] == hostgroup_name:
-                return hostgroup.get ('id')
+        logging.debug('Reached get_hostgroup_id_v1')
+        hostgroups = self.get("hostgroups", headers=self.headers).json()["results"]
+        hostgroups = list(filter(lambda x: x.get("name")==hostgroup_name, hostgroups))
```
Comment (ivansa-ru) [inline oc_cdtapi/ForemanAPI.py:793]: This writing does not add readability and complicates code maintenance

File: setup.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 from setuptools import setup
 from sys import version_info
 
-__version = "3.9.2"
+__version = "3.9.3"
```
Comment (ivansa-ru) [inline setup.py:6]: Please increase version. I see a number of commits since last successful deploy.

File: oc_cdtapi/ForemanAPI.py (hunk 1 above)
Comment (vifrrg) [reply oc_cdtapi/ForemanAPI.py]: That was not my change, but adjustded.

File: oc_cdtapi/ForemanAPI.py (hunk 2 above)
Comment (vifrrg) [reply oc_cdtapi/ForemanAPI.py]: Disagree that three-lined loop with different indent is more readable than one simple `lambda filter`.
But reverted to (almost) previous version.

File: setup.py (hunk 3 above)
Comment (vifrrg) [reply setup.py]: Done, 3.9.4

**Review by ivansa-ru — APPROVED** [review]

**Pull request merged**